tests/test_basic_relay.go

```diff
@@ -0,0 +1,298 @@
+		relayerCancel       context.CancelFunc
+	)
+
+	ginkgo.It("Set up relayer config", ginkgo.Label("Relayer", "Setup Relayer"), func() {
+		hostA, portA, err := teleporterTestUtils.GetURIHostAndPort(teleporterTestUtils.ChainANodeURIs[0])
+		Expect(err).Should(BeNil())
+
+		hostB, portB, err := teleporterTestUtils.GetURIHostAndPort(teleporterTestUtils.ChainBNodeURIs[0])
+		Expect(err).Should(BeNil())
+
+		log.Info(
+			"Setting up relayer config",
+			"hostA", hostA,
+			"portA", portA,
+			"blockChainA", teleporterTestUtils.BlockchainIDA.String(),
+			"hostB", hostB,
+			"portB", portB,
+			"blockChainB", teleporterTestUtils.BlockchainIDB.String(),
+			"teleporterTestUtils.SubnetA", teleporterTestUtils.SubnetA.String(),
+			"teleporterTestUtils.SubnetB", teleporterTestUtils.SubnetB.String(),
+		)
+
+		relayerConfig := config.Config{
+			LogLevel:          logging.Info.LowerString(),
+			NetworkID:         peers.LocalNetworkID,
+			PChainAPIURL:      teleporterTestUtils.ChainANodeURIs[0],
+			EncryptConnection: false,
+			StorageLocation:   storageLocation,
+			SourceSubnets: []config.SourceSubnet{
+				{
+					SubnetID:          teleporterTestUtils.SubnetA.String(),
+					ChainID:           teleporterTestUtils.BlockchainIDA.String(),
+					VM:                config.EVM.String(),
+					EncryptConnection: false,
+					APINodeHost:       hostA,
+					APINodePort:       portA,
+					MessageContracts: map[string]config.MessageProtocolConfig{
+						teleporterTestUtils.TeleporterContractAddress.Hex(): {
+							MessageFormat: config.TELEPORTER.String(),
+							Settings: map[string]interface{}{
+								"reward-address": teleporterTestUtils.FundedAddress.Hex(),
+							},
+						},
+					},
+				},
+			},
+			DestinationSubnets: []config.DestinationSubnet{
+				{
+					SubnetID:          teleporterTestUtils.SubnetB.String(),
+					ChainID:           teleporterTestUtils.BlockchainIDB.String(),
+					VM:                config.EVM.String(),
+					EncryptConnection: false,
+					APINodeHost:       hostB,
+					APINodePort:       portB,
+					AccountPrivateKey: teleporterTestUtils.FundedKeyStr,
+				},
+			},
+		}
+
+		data, err := json.MarshalIndent(relayerConfig, "", "\t")
+		Expect(err).Should(BeNil())
+
+		f, err := os.CreateTemp(os.TempDir(), "relayer-config.json")
+		Expect(err).Should(BeNil())
+
+		_, err = f.Write(data)
+		Expect(err).Should(BeNil())
+		relayerConfigPath = f.Name()
+
+		log.Info("Created awm-relayer config", "configPath", relayerConfigPath, "config", string(data))
+	})
+
+	ginkgo.It("Build Relayer", ginkgo.Label("Relayer", "Build Relayer"), func() {
+		// Build the awm-relayer binary
+		cmd := exec.Command("./scripts/build.sh")
+		out, err := cmd.CombinedOutput()
+		fmt.Println(string(out))
+		Expect(err).Should(BeNil())
+	})
+
+	// Send a transaction to Subnet A to issue a Warp Message from the Teleporter contract to Subnet B
+	ginkgo.It("Send Message from A to B", ginkgo.Label("Warp", "SendWarp"), func() {
```

Comment:
nit: can we change label to Teleporter instead of Warp

Reply:
Renamed `"Relayer", "SendTeleporter"`